docs/REST/TEC/V1/abstract-classes.md

```diff
@@ -1,147 +1,464 @@
+- **404 Handling**: Automatic handling of missing entities
+
+#### Required Methods
+
+```php
+abstract public function get_post_type(): string;
+abstract public function get_model_class(): string;
+abstract public function guest_can_read(): bool;
+```
 
 #### Key Methods
 
-- `guest_can_read()` - Whether guests can read (default: false)
-- `can_read()` - Permission check for reading, relies on `WP_REST_Posts_Controller` for permissions around the Post Type, password protection and status protections.
-- `can_create()` - Permission check for creating
-- `can_update()` - Permission check for updating
-- `can_delete()` - Permission check for deleting
-- `get_post_type_object()` - Gets the WordPress post type object
-- `format_post_entity_collection()` - Formats multiple posts
-- `get_formatted_entity()` - Formats a single post
-- `add_properties_to_model()` - Adds model properties to response
-- `validate_status()` - Validates post status values
-- `transform_entity()` - Transform hook for entities
-
-#### Constants
-
-- `ALLOWED_STATUS` - Array of allowed post statuses:
-  - `publish`
-  - `pending`
-  - `draft`
-  - `future`
-  - `private`
+- `get_post(int $id)`: Retrieve a post by ID
+- `format_post(WP_Post $post)`: Format post for response
+- `can_read(WP_REST_Request $request)`: Check read permissions based on post capabilities
+- `can_create(WP_REST_Request $request)`: Check create permissions
+- `can_update(WP_REST_Request $request)`: Check update permissions for specific post
+- `can_delete(WP_REST_Request $request)`: Check delete permissions for specific post
+- `guest_can_read()`: Abstract method to define public access
 
 #### Usage Example
 
 ```php
-class EventEndpoint extends Post_Entity_Endpoint implements RUD_Endpoint {
+class Event extends Post_Entity_Endpoint implements RUD_Endpoint {
     public function get_post_type(): string {
         return 'tribe_events';
     }
-
+    
     public function get_model_class(): string {
-        return Event_Model::class;
+        return \Tribe\Events\Models\Post_Types\Event::class;
     }
-
+    
     public function guest_can_read(): bool {
         return true; // Public events
     }
+}
+```
+
+### `Definition`
+
+**Location**: `Definition.php`
+
+Base class for OpenAPI schema definitions.
+
+#### Key Features
+
+- **Schema Generation**: Generates OpenAPI-compliant schemas
+- **Composition Support**: Supports allOf pattern for inheritance
+- **Type System**: Integrates with parameter type system
+- **Priority System**: Controls registration order
+
+#### Required Methods
+
+```php
+abstract public function get_name(): string;
+abstract public function get_definition(): array;
+```
+
+#### Key Methods
+
+- `get_priority()`: Returns registration priority
+- `to_array()`: Converts to OpenAPI schema array
+
+#### Usage Example
 
-    // Implement RUD_Endpoint methods...
+```php
+class Event_Definition extends Definition {
+    public function get_name(): string {
```

Comment:
One other nitpick: since these examples are in the context of WordPress, they should use tabs instead of spaces for indenting.